torch/csrc/autograd/python_cpp_function.cpp

```diff
@@ -255,11 +255,9 @@ struct DefaultFunctionType {
   }
 
   PyTypeObject type;
-};
+} default_type;
```

Comment:
This doesn't look right. You're static initializing default_type now, how do you guarantee you have the GIL?

Reply:
Sam's answer was not a straight yes so updated.